File: test/authorization-profile.card.test.js
'use strict';

const assert = require('assert');
const { authProfiles } = require('../src/config/authProfiles');
const { authorizePayment } = require('../src/services/authorizationService');

/**
 * Visa and Mastercard profile-selection coverage. Both networks are certified against
 * the shared default profile.
 */

assert.strictEqual(authProfiles.forNetwork('visa').id, 'card-default');
assert.strictEqual(authProfiles.forNetwork('mastercard').id, 'card-default');

for (const network of ['visa', 'mastercard']) {
  const record = authorizePayment({
    paymentId: `pay_${network}_1`,
    invoiceId: `inv_${network}_1`,
    network,
    amountMinor: 2500,
    currency: 'USD',
  });
  assert.strictEqual(record.profileId, 'card-default');
  assert.strictEqual(record.status, 'authorized');
}

process.stdout.write('authorization-profile.card.test.js passed\n');

File: src/test/kotlin/com/beaconstone/payments/AuthorizationProfileCardTest.kt
package com.beaconstone.payments

import com.beaconstone.payments.config.authProfiles
import com.beaconstone.payments.services.Payment
import com.beaconstone.payments.services.authorizePayment
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

/**
 * Card-network profile-selection coverage.
 *
 * Authorisation resolves the default profile, which every supported network is
 * certified against.
 */
class AuthorizationProfileCardTest {
    @Test
    fun `default profile is card-default`() {
        assertEquals("card-default", authProfiles.default.id)
    }

    @Test
    fun `visa and mastercard authorise against the default profile`() {
        for (network in listOf("visa", "mastercard")) {
            val record = authorizePayment(
                Payment(
                    paymentId = "pay_${network}_1",
                    invoiceId = "inv_${network}_1",
                    network = network,
                    amountMinor = 2500,
                    currency = "USD",
                ),
            )
            assertEquals("card-default", record.profileId)
            assertEquals("authorized", record.status)
        }
    }
}
